resources/testharness.js

```diff
@@ -937,7 +956,17 @@ policies and contribution forms [3].
                                         String(e),
                                         e && e.stack);
                        tests.complete();
-                   });
+                   })
+            .finally(function()
+                {
+                    tests.promise_setup_pending--;
```

Comment:
(This might be identical to, or at least impacted by, the discussion above about whether to create a new `Test` object before deferring the rest of the work, I haven't thought it through sufficiently.)

I'm not sure if we want to execute this block if we threw an error; `tests.complete()` indicates the end of running tests and trying to run them after that call seems strange.

Reply:
I believe your are right. I moved it out of the finally block.